metashare/api/gh.py

```diff
@@ -114,15 +114,30 @@ def local_github_checkout(user, repo_id, commit_ish=None):
             yield repo_root
 
 
-def get_cumulus_prefix(**kwargs):
+def get_project_config(**kwargs):
     """
     Expects to be in a local_github_checkout.
     """
     global_config = GlobalConfig()
-    project_config = global_config.get_project_config(**kwargs)
+    return global_config.get_project_config(**kwargs)
+
+
+def get_cumulus_prefix(**kwargs):
+    """
+    Expects to be in a local_github_checkout.
+    """
+    project_config = get_project_config(**kwargs)
     return project_config.project__git__prefix_feature
 
 
+def get_source_format(**kwargs):
+    """
+    Expects to be in a local_github_checkout.
+    """
+    project_config = get_project_config(**kwargs)
+    return project_config.project__source_format
```

Comment:
@wlonk It's not clear to me what the difference is (or which is preferred) between this use of `GlobalConfig` vs the use of `BaseCumulusCI` in `sf_org_changes.run_retrieve_task`. It seems like both end up with the project config (`global_config.get_project_config()` vs `cci.project_config`).

Reply:
cc @davisagli -- you might have input on the above question.

Reply:
@davisagli can answer that better than I can.

Reply:
BaseCumulusCI gives you a "CumulusCI runtime" object which provides access to the global config (representing the contents of cumulusci's own cumulusci.yml), the project config (representing the contents of a specific project's cumulusci.yml), and a keychain (which is meant to be an abstraction for accessing org configs regardless of whether they're in cumulusci.yml or on the filesystem or somewhere else, but MetaShare is only using the keychain as a shortcut to construct an org config that's defined in cumulusci.yml)

GlobalConfig is just the global config, but it also provides `get_project_config` as a shortcut to construct the project config.

The default BaseGlobalConfig and BaseProjectConfig classes defined in CumulusCI (which are the ones that you get if you use BaseCumulusCI) do some things that MetaShare doesn't need or want. Specifically they check for overrides in the user's ~/.cumulusci directory (which is relevant for CLI use but not in a web app) and they use git to determine the project's git URL, branch and commit if not passed explicitly (which MetaShare should generally be able to do). So for MetaShare we created special GlobalConfig and ProjectConfig subclasses in metashare/api/custom_cci_configs.py that avoid that logic.

In order to get the special classes you currently have to construct them directly (or construct GlobalConfig and then use its get_project_config to construct a ProjectConfig). It might make sense to define a MetashareCumulusCIRuntime as a subclass of BaseCumulusCI to provide access to all the configs with an API that's more similar to BaseCumulusCI.

Reply:
@davisagli That would make sense to me. Since this PR is working (and not introducing a new way for MS to access package configs), I've created a new story for that cleanup: https://trello.com/c/BtES8zHy